duckduckgo/Android
# Implementation task for Downloads UI [no modules]

Task/Issue URL: https://app.asana.com/0/0/1201892740921878/f

### Description
Added downloads UI. Fixed images downloaded as text. Fixed apk mime type. Updated tests.

### Steps to test this PR

- Install from this branch
- Download files (you can use this website or a similar one -> https://file-examples.com/)
- Check the below:
        - when download starts you see a snackbar mentioning this
        - when download finishes you see a snackbar mentioning this and an action to open the file
        - you can open the file or see an error if there is not app that can handle the specific file
        - if you download the same file multiple times the file name contains a "-x" before the extension, where x is an incremental number
- open the Downloads view from the menu
- Check the below:
        - you see the list of the downloaded files
        - for each file you see the name and the size. If the file is still downloading you see "In Progress" instead of the size
        - the list contains headers like "TODAY", "YESTERDAY", "PAST WEEK" etc. You can alter the download date (createdAt) in the database (app.db -> downloads) to simulate various scenarios.
        - tap on an item in the list and you should be able to open the file or see an error if there is not app that can handle the specific file
        - tap on the overflow menu for an item and share it
        - tap on the overflow menu for an item and delete it. You should see a snackbar with an action to UNDO. Both delete and undo should work properly.
        - notice there is a search in the toolbar. Try to search for an existing file / a file that doesn't exist and see how it works.
        - tap on the overflow menu to the whole screen and delete all files. You should see a snackbar with an action to UNDO. Both delete and undo should work properly.
        - when no files are downloaded (or they were deleted) you should see an empty view
        - try to download a large file (you can use this website or a similar one -> https://www.thinkbroadband.com/download).
        - while the file is downloading try to tap on its item and notice is not clickable.

### UI changes

![Screenshot_20220405_180737](https://user-images.githubusercontent.com/7963079/161812067-9e876aeb-c3c2-49e7-9f91-ef4d6221cb3c.png)
